src/compas/geometry/transformations/quaternions.py

```diff
@@ -0,0 +1,339 @@
+"""
+This module contains functions that operate on and/or return quaternions.
+
+Notes
+-----
+The default convention to represent a quaternion :math:`q` in this module is by four real values **w**, **x**, **y**, **z**.
+The first value **w** is the scalar (real) part, and **x**, **y**, **z** form the vector (complex, imaginary) part [1]_, so that:
+
+:math:`q = w + xi + yj + zk`
+
+where :math:`i, j, k` are basis components with following multiplication rules [2]_:
+
+:math:`ii = jj = kk = ijk = -1`
```

Comment:
use `.. math::` for blocks of math stuff. `:math:` is for inline math formulas...